src/types/frozenset.rs

```diff
@@ -0,0 +1,194 @@
+// Copyright (c) 2017-present PyO3 Project and Contributors
+//
+
+use crate::err::{PyErr, PyResult};
+#[cfg(Py_LIMITED_API)]
+use crate::types::PyIterator;
+use crate::{ffi, AsPyPointer, PyAny, Python, ToPyObject};
+
+use std::ptr;
+
+/// Represents a  Python `frozenset`
+#[repr(transparent)]
+pub struct PyFrozenSet(PyAny);
+
+pyobject_native_type!(
+    PyFrozenSet,
+    ffi::PySetObject,
+    ffi::PyFrozenSet_Type,
+    #checkfunction=ffi::PyFrozenSet_Check
+);
+
+impl PyFrozenSet {
+    /// Creates a new frozenset.
+    ///
+    /// May panic when running out of memory.
```

Comment:
I know not from this PR, but is this actually true? It looks like returning `PyResult` means that we'll actually return `PyErr` containing a `MemoryError` if we run out of memory.

I suspect that actually the text may want to be more like `PySet` where we don't mention this and instead comment on the error return being caused by unhashable contents.